### PR TITLE
new recording

### PR DESCRIPTION
**Pull request by koldn**

### CODE REVIEW

File: recorder/src/main/resources/identity.js [hunk 1]
```diff
@@ -14,6 +14,29 @@ jsflight.getElementXPath = function(element) {
         return jsflight.getElementTreeXPath(element);
 };
 
+jsflight.getElementXpathId = function(element){
```
Comment (d0k1) [inline recorder/src/main/resources/identity.js:21]: It might be better to save all attributes of the element. Player can easily skip unused ones

**Comment by d0k1** [comment]
@koldn  you were supposed to deal with clicks to "nowhere" / "black hole" element. But I cold not find a solution for that case in this pull request. Did you do it?

**Comment by koldn** [comment]
Not yet, because of scenario adaptation priority

File: player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java [hunk 2]
```diff
@@ -507,7 +517,20 @@ private static void makeElementVisibleByJS(JSONObject event, String target)
 
     private static void resizeForEvent(WebDriver wd, JSONObject event)
     {
-        wd.manage().window().setSize(new Dimension(event.getInt("window.width"), event.getInt("window.height")));
+        int w = 1920;
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: 1000 x 1000

File: recorder/src/main/resources/options.js [hunk 3]
```diff
@@ -35,6 +35,10 @@ jsflight.options = {
     send_interval : -1,
     // control panel disabled by default
     cp_disabled : true,
+    // list of element`s attributes to check and store
```
Comment (d0k1) [inline recorder/src/main/resources/options.js]: example

File: recorder/src/main/resources/identity.js [hunk 4]
```diff
@@ -14,6 +14,33 @@ jsflight.getElementXPath = function(element) {
         return jsflight.getElementTreeXPath(element);
 };
 
+jsflight.checkIdIsNotExcluded = function(id){
+  var reg = new RegExp(jsflight.options.id_exclusions.join('|'));
```
Comment (d0k1) [inline recorder/src/main/resources/identity.js]: it is a constant!

File: recorder/src/main/resources/identity.js [hunk 5]
```diff
@@ -14,6 +14,33 @@ jsflight.getElementXPath = function(element) {
...
+    for (; element && element.nodeType == 1; element = element.parentNode) {
+        var attr = element.attributes;
+        var idr = element.id;
+        var tag = element.tagName.toLowerCase();
+        if(!idr && tag == "input"){
+            paths.push("//"+tag);    
+        };
+        if(idr && jsflight.checkIdIsNotExcluded(idr)){
+            paths.splice(0,0,"//*[@id='"+idr+"']");
+        }
```
Comment (d0k1) [inline recorder/src/main/resources/identity.js]: use else

**Pull request merged**